test/azure/generated/AzureCompositeModelClient/models/index.ts

```diff
@@ -184,11 +184,20 @@ export interface Siamese extends Cat {
   breed?: string;
 }
 
+/**
+ * Contains the possible cases for Fish.
+ */
+export type FishUnion = Fish | SalmonUnion | SharkUnion;
+
 /**
  * @interface
  * An interface representing Fish.
  */
 export interface Fish {
+  /**
+   * @member {string} fishtype Polymorphic Discriminator
+   */
+  fishtype: "Fish";
```

Comment:
I inserted a .OrderBy in the generator to make the discriminator always the first property in the polymorphic interface. I think this is a better user experience.